### ISSUE TITLE
Not reporting incorrect return type 

### ISSUE THREAD

**Issue by vivax3794** [microsoft/pyright#2363]
**Describe the bug**
pyright is not reporting wrong return type when using a `TypeVar` of `int` and a `int` subclass

**To Reproduce**
```python
from typing import TypeVar

class Foo(int):
    abc = 4

T = TypeVar("T", int, Foo)

def add(value: T) -> T:
    reveal_type(value + 5) # int
    # but pyright is not complaing
    return value + 5 

# errors at runtime becuase add(Foo(5)) returns an int
add(Foo(5)).abc
```

**Expected behavior**
An error explaining the returned type is not correct like: ` Expression of type "int" cannot be assigned to return type "T@add"` 

currently only gives
```
main.py:11:17 - info: Type of "value + 5" is "int*"
```

**command-line**
Version: 1.1.173
`typeCheckingMode = "strict"`

**Comment by erictraut** [comment]
Thanks for the bug report. I agree this is a false negative — it should generate an error in this case. I'll investigate further.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.174, which I just published. It will also be included in the next release of pylance.
